elastic/kibana
# [ResponseOps][Actions] Make actions retry when encountering failures

Resolves https://github.com/elastic/kibana/issues/143046

## Summary

Updated `maxAttempts` for all connectors to be 3. After looking into the error handling for the post-processing after action execution, it looks like [we won't retry if the execution is successful and one of the async processes fails](https://github.com/elastic/kibana/blob/main/x-pack/plugins/actions/server/lib/task_runner_factory.ts#L166). 

Please let me know if there is another post-processing place that I should update.

### To verify

- Create a rule with a connector.
- To cause it to fail, I usually force the return of an error status with retry set in the connectors executor
A good example of this is in `x-pack/plugins/stack_connectors/server/connector_types/stack/slack/index.ts`
- Verify that it will only retry 3 times
